src/test/yield.t.sol

```diff
@@ -0,0 +1,78 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.16;
+
+import "forge-std/Test.sol";
+import "forge-std/console.sol";
+
+// import all major contracts
+import "../Converter.sol";
+import "../Lender.sol";
+import "../Creator.sol";
+import "../ETHWrapper.sol";
+import "../Redeemer.sol";
+ 
+// import adapters
+import "../adapters/YieldAdapter.sol"; 
+
+contract YieldTest is Test {
+
+    address USDC = 0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48;
+
+    uint256 maturity = 1704975690;
+
+    address yieldDecember = 0x9536C528d9e3f12586ea3E8f624dACb8150b22aa;
+
+    address userPublicKey = 0x3f60008Dfd0EfC03F476D9B489D6C5B13B3eBF2C;
+
+    uint256 startingBalance = 100000 ether;
+
+    function setUp() public {
+
+        // Deploy all major contracts
+        Creator creator = new Creator();
+        Converter converter = new Converter();
+        ETHWrapper ethWrapper = new ETHWrapper();
+        Lender lender = new Lender(address(0), address(0), address(0));
+        Redeemer redeemer = new Redeemer(address(lender), address(0), address(0));
+        MarketPlace marketplace = new MarketPlace(address(redeemer), address(lender), address(creator));
```

Comment:
I don't quite understand this part here: We're passing the lender and redeemer address to the marketplace constructor, but we're still calling `setLender` and `setRedeemer` on the marketplace later?

Reply:
lol yeah youre right -- its redundant, i was just double checking all methods and the workflow now that each contract has the same storage (there will need to be a couple more setters most likely)